src/Storages/System/StorageSystemErrors.cpp

```diff
@@ -10,30 +11,44 @@ namespace DB
 NamesAndTypesList StorageSystemErrors::getNamesAndTypes()
 {
     return {
-        { "name",              std::make_shared<DataTypeString>() },
-        { "code",              std::make_shared<DataTypeInt32>() },
-        { "value",             std::make_shared<DataTypeUInt64>() },
+        { "name",                    std::make_shared<DataTypeString>() },
+        { "code",                    std::make_shared<DataTypeInt32>() },
+        { "value",                   std::make_shared<DataTypeUInt64>() },
+        { "last_error_time",         std::make_shared<DataTypeDateTime>() },
+        { "last_error_message",      std::make_shared<DataTypeString>() },
+        { "last_error_stacktrace",   std::make_shared<DataTypeString>() },
+        { "remote",                  std::make_shared<DataTypeUInt8>() },
     };
 }
 
 
 void StorageSystemErrors::fillData(MutableColumns & res_columns, const Context & context, const SelectQueryInfo &) const
 {
-    for (size_t i = 0, end = ErrorCodes::end(); i < end; ++i)
+    auto add_row = [&](std::string_view name, size_t code, size_t value, UInt64 error_time_ms, const std::string & message, const std::string & stacktrace, bool remote)
     {
-        UInt64 value = ErrorCodes::values[i];
-        std::string_view name = ErrorCodes::getName(i);
-
-        if (name.empty())
-            continue;
-
         if (value || context.getSettingsRef().system_events_show_zero_values)
         {
             size_t col_num = 0;
             res_columns[col_num++]->insert(name);
-            res_columns[col_num++]->insert(i);
+            res_columns[col_num++]->insert(code);
             res_columns[col_num++]->insert(value);
+            res_columns[col_num++]->insert(error_time_ms / 1000);
+            res_columns[col_num++]->insert(message);
+            res_columns[col_num++]->insert(stacktrace);
+            res_columns[col_num++]->insert(remote);
         }
+    };
+
+    for (size_t i = 0, end = ErrorCodes::end(); i < end; ++i)
+    {
+        const auto & error = ErrorCodes::values[i].get();
+        std::string_view name = ErrorCodes::getName(i);
+
+        if (name.empty())
+            continue;
+
+        add_row(name, i, error.local,  error.error_time_ms, error.message, error.stacktrace, false /* remote=0 */);
+        add_row(name, i, error.remote, error.error_time_ms, error.message, error.stacktrace, true  /* remote=1 */);
```

Comment:
`error.message` and `error.stacktrace` will be the same for local and remote case, cause that's how we store it in `ErrorCodes::values`. So, it will not be correct for one of rows.

Reply:
Indeed, nice catch, fixed